src/dso_api/dynamic_api/views/api.py

```diff
@@ -228,3 +232,69 @@ class CustomViewSet(DynamicApiViewSet):
         "table_id": model._table_schema["id"],
     }
     return type(f"{model.__name__}ViewSet", (DynamicApiViewSet,), attrs)
+
+
+class APIIndexView(APIView):
+    """An overview of API endpoints for a list of Datasets,
+    in a JSON format compatible with developer.overheid.nl.
+    """
+
+    schema = None  # exclude from schema
+
+    # For browsable API
+    name = "DSO-API"
+    description = (
+        "To use the DSO-API, see the documentation at <https://api.data.amsterdam.nl/v1/docs/>. "
+    )
+
+    # Set by as_view
+    api_type = "rest_json"
+
+    def get_datasets(self) -> List[Dataset]:
+        return list(get_published_datasets().order_by("name"))
+
+    def get_environments(self, ds: Dataset, base: str) -> List[dict]:
+        return [
+            {
+                "name": "production",
+                "api_url": base + reverse(f"dynamic_api:openapi-{ds.schema.id}"),
+                "specification_url": "",
+                "documentation_url": "",
+            }
+        ]
+
+    def get_related_apis(self, ds: Dataset, base: str) -> List[dict]:
+        """ Get list of other APIs exposing the same dataset """
+        return []
+
+    def get(self, request, *args, **kwargs):
+        base = request.build_absolute_uri("/").rstrip("/")
```

Comment:
wat je kan overwegen:

```suggestion
        self.base_url = request.build_absolute_uri("/").rstrip("/")
```

en dat overal in de methoden te gebruiken.

Reply:
Or introduce the `URL` class from https://pypi.org/project/more-ds/ . We already use that in `schema-tools` and `dataservices-airflow`. It obviates the need for worrying about the existence or lack of an ending `/` of an URL. It also allows for easy URL concatenation.

Having said that it doesn't do any validation.